### ISSUE TITLE
ubuntu ppa uptodate

### ISSUE THREAD

**Issue by ramonskie** [direnv/direnv#299]
the ubuntu direnv versions and all its ppa's have still an old direnv version
it would be nice if this could uptodate

**Comment by zimbatm** [comment]
hi, I don't have control over the PPA so it's probably best to contact them directly.

As a workaround you can [download direnv binary from the latest release](https://github.com/direnv/direnv/releases). `chmod +x direnv` it and place it in /usr/local/bin.

**Comment by haizaar** [comment]
This one seem yours, isn't it? https://launchpad.net/~zimbatm/+archive/ubuntu/direnv
Do you have any plans to populate it with packages?

Downloading binary is nice, but then I'm missing updates.

Thank you!

**Comment by zimbatm** [comment]
Nope. Debian packaging is too much work and I don't use the platform.

I deleted that PPA since it's unsupported and didn't contain any releases.

**Comment by G-Rath** [comment]
Any update or change on this?

I'm not too familiar w/ the linux/apt package system, but would love to get the latest version of direnv published for ubuntu/bionic if possible, as that's what `wsl` is semi-officially locked into atm.

(You can technically update in a few ways, but iirc it's not been released on official channels so that Windows 10 natively offers it - it will "soon", but WSl2 is the main focus right now).

I see that the bionic package is coming from [here](https://github.com/punitagrawal/direnv), while the later ubuntu versions are from direnv directly [here](https://github.com/direnv/direnv-deb).

Looking at the `direnv-deb` repo, it looks as if the bionic repo was "handed over", as the commits are all from @punitagrawal 

Apologies in advance if I've misunderstood something or used the wrong term - I'm very new to the linux/ubuntu package system, so part of this is just wanting to try and get a better understanding :)

**Comment by punitagrawal** [comment]
Hi Gareth,

Hopefully I can help answer your query.

I maintain the debian package for direnv. This roughly translates to taking
the upstream direnv releases semi-regularly and packaging it up for debian.
You can follow the debian package tracker at -
https://tracker.debian.org/pkg/direnv

The ubuntu package for direnv is a snapshot of the debian package.

The debian package has been lagging upstream as we've been in a freeze for
Buster release since the beginning of the year. Now that Buster is released
I'll try and get the latest version packaged in the coming days.

Having said that, I am not sure about the process to upgrade a version of a
package in an older version of Ubuntu. But upgrading the debian package
will ensure that the next Ubuntu release will pick it up.

Hope that helps.

**Comment by G-Rath** [comment]
Thanks for taking the time to reply :)

That all makes rough sense to me - if my understanding is correct, the thing reason (aside from the Buster freeze) is that the `direnv-deb` repo is for building against the latest version of debian, and so produces a single build that is only compatible w/ certain versions (i.e "the last three ubuntu releases), rather than producing a build for each major version of choice?

It's not a huge deal as the core of `direnv` works just fine (and btw is fantastic); I'll just have to wait until MS release WSL2 & Cosmic :)

**Comment by zimbatm** [comment]
@punitagrawal thanks by the way for doing all the packaging work!

**Comment by punitagrawal** [comment]
Of course - packaging is the easy part. Thanks for all your effort on
making such a useful tool! :)
